bookmarks/vanilla/test/unit/Bookmark.test.ts

```diff
@@ -29,135 +29,236 @@ describe("Bookmark", () => {
+        mockFetch.mockRestore();
     });
-    it("the get function should get one bookmark", async () => {
+
+    it("the delete function should delete bookmark", async () => {
         const defaultBookmarksDocUrl = "https://fake-pod.net/path/to/bookmark-formats.ttl";
-        const url = 'https://fake-pod.net/path/to/bookmark-formats.ttl#d2d50f70-8eb0-40b6-9996-88c4a430a16d';
+        const url = 'https://fake-pod.net/path/to/bookmark-formats.ttl#:one';
 
-        const expected = {
-            url: 'https://fake-pod.net/path/to/bookmark-formats.ttl#d2d50f70-8eb0-40b6-9996-88c4a430a16d',
-            title: 'updated',
-            link: 'http://goo.com'
+        const responseGet: any = {
+            status: 200,
+            ok: true,
+            headers: {
+                get: (h: string) => (h == "Content-Type" ? "text/turtle" : undefined)
+            },
+            text: () => {
+                return Promise.resolve(loadFixture("bookmark-formats.ttl"));
+            }
+        };
+
+        const responseDelete = {
+            ...responseGet,
+            text: () => {
+                return Promise.resolve(loadFixture("bookmark-formats-without-one.ttl"));
+            }
         }
 
-        jest.spyOn(Bookmark, "getIndexUrl").mockReturnValue(Promise.resolve(defaultBookmarksDocUrl));
+        const mockGetIndexUrl = jest.spyOn(Bookmark, "getIndexUrl").mockReturnValue(Promise.resolve(defaultBookmarksDocUrl));
 
-        jest.spyOn(inruptClient, "getSolidDataset").mockReturnValue(Promise.resolve(JSON.parse(loadFixture("ds.json"))));
+        const mockFetchGet = jest.spyOn(session, "fetch").mockReturnValue(Promise.resolve(responseGet));
+        const mockGetThing = jest.spyOn(inruptClient, "getThing").mockReturnValue(JSON.parse(loadFixture("things/one.json")));
+        const mockFetchDelete = jest.spyOn(session, "fetch").mockReturnValue(Promise.resolve(responseDelete));
 
-        const res = await Bookmark.get(url, session);
+        const res = await Bookmark.delete(url, session);
 
         expect(Bookmark.getIndexUrl).toHaveBeenCalled();
 
-        expect(res).toEqual(expected);
+        expect(res).toEqual(true);
+        mockGetIndexUrl.mockRestore();
+        mockFetchGet.mockRestore();
+        mockGetThing.mockRestore();
+        mockFetchDelete.mockRestore();
     });
-    it("the delete function should delete bookmark", async () => {
-        const defaultBookmarksDocUrl = "https://fake-pod.net/path/to/bookmark-formats.ttl";
-        const url = 'https://fake-pod.net/path/to/bookmark-formats.ttl#d2d50f70-8eb0-40b6-9996-88c4a430a16d';
-
-        const expected = [{
-            url: 'https://fake-pod.net/path/to/bookmark-formats.ttl#d2d50f70-8eb0-40b6-9996-88c4a430a16d',
-            title: 'updated',
-            link: 'http://goo.com'
-        }]
 
-        jest.spyOn(Bookmark, "getIndexUrl").mockReturnValue(Promise.resolve(defaultBookmarksDocUrl));
+    it("the create function should create bookmark", async () => {
```

Comment:
Grammar: The create **method** should create **a** bookmark

Reply:
done